File: 이상욱/BOJ/m4w10/BOJ1931.java
package m4w10;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;

public class BOJ1931 {
	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		
		int[][] map = new int[n][2];
		
		StringTokenizer st;
		
		for(int i = 0; i < n; i++) {
			st = new StringTokenizer(br.readLine());
			map[i][0] = Integer.parseInt(st.nextToken());
			map[i][1] = Integer.parseInt(st.nextToken());
			
		}
		
		Arrays.sort(map,new Comparator<int[]>() {
			
			@Override
			public int compare(int[] arr1, int[] arr2) {
				
				if(arr1[1] == arr2[1]) {
					return arr1[0] - arr2[0];
				}

				return arr1[1] - arr2[1];
			}
		});
		
		int cnt = 0;
		int end = 0;
		
		for(int i = 0; i < map.length; i++) {
			
			if(end <= map[i][0]) {
				end = map[i][1];
				cnt++;
			}
			
		}
		System.out.println(cnt);
		
	}
}

File: 이상욱/BOJ/m4w10/BOJ1931.kt
package m4w10

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main(){

    val br = BufferedReader(InputStreamReader(System.`in`))

    val n = br.readLine().toInt()

    val map = Array(n){IntArray(2)}
    var st : StringTokenizer

    for(i in 0 until n){
        st = StringTokenizer(br.readLine())
        map[i][0] = st.nextToken().toInt()
        map[i][1] = st.nextToken().toInt()
    }

    map.sortWith(compareBy({it[1]},{it[0]}))

    var cnt = 0
    var end = 0

    map.forEach {
        if(end <= it[0]){

            end = it[1]
            cnt++
        }
    }

    println(cnt)

}